### ISSUE TITLE
Exercise 2.1

### ISSUE THREAD

**Issue by michaelwayman** [mharbuz/rlbook-exercises#10]
> n ε-greedy action selection, for the case of two actions and ε = 0.5, what is the probability that the greedy action is selected?
> 
> Answer
> Greedy action will be seleted with probability equals 50%.
> 
@mharbuz I will probably be commenting/asking several questions the next couple months. I'm just beginning to work through this book and I appreciate your repo here, you are further ahead than me for sure.

But I wonder if 2.1 might be a trick question. If so, the probability of choosing the greedy action is 50% yes, but then even if we take the random action there is another 50% chance that we'll choose the greedy action.
So I think in practice if epsilon is 0.5 and the greedy action is fixed, we'll end up taking the greedy action 75% of the time.

```python
import numpy as np

epsilon = 0.5
q_values = [1, 0]
action_counts = [0, 0]
greedy_action = np.argmax(q_values)

for _ in range(10000):
    if 1 - epsilon > np.random.random():
        action_to_take = greedy_action
    else:
        action_to_take = np.random.choice(q_values)
    action_counts[action_to_take] += 1

greedy_percent = action_counts[greedy_action] / sum(action_counts) * 100
print(f"Greedy action was taken {greedy_percent}% of the time")
```

```
output: Greedy action was taken 74.73% of the time
```

**Comment by gblawrence03** [comment]
This is correct. It's the same reason why, on page 30, it says "The e = 0.1 method explored more, and usually found the optimal action earlier, but it never selected that action more than 91% of the time". With e = 0.1, the optimal action is selected with probability 0.9, and then the remaining 10% of the time, the optimal action is selected with probability 0.1 (because there are 10 actions). So the probability = 0.9 + 0.1 * 0.1 = 0.91. 

More generally, the probability of selecting the optimal action is (1 - e) + e * (1 / k), where k is the number of actions to choose from, assuming that there is only one optimal action at any given time step.
